text_generation/causal_lm/cpp/beam_search_causal_lm.cpp

```diff
@@ -22,6 +22,14 @@ std::string detokenize(ov::InferRequest& detokenizer, const std::vector<int64_t>
     detokenizer.infer();
     return detokenizer.get_output_tensor().data<std::string>()[0];
 }
+
+void set_state_max_batch(std::vector<ov::VariableState>&& states, size_t batch_size) {
+    for (ov::VariableState& state : states) {
+        ov::Shape shape = state.get_state().get_shape();
+        shape.front() = batch_size;
+        state.set_state(ov::Tensor{ov::element::f32, shape});
+    }
+}
```

Comment:
Does it mean that reshape doesn't work for the model?

Is this a workaround for broken dynamic batch size?

Reply:
It's not a full reshape. It's an implementation of suggested workaround to reshape states only, to fix broken dynamic batch size.

You can see in the tests status, that it didn't work out.